### PR TITLE
Add a safe API for user data

### PR DESCRIPTION
**Pull request by SimonSapin**

### CODE REVIEW

**Comment by SimonSapin** [comment]
Should doc-comments go into https://github.com/gtk-rs/lgpl-docs? Are docs written specifically for Rust bindings also under the LGPL?

**Comment by GuillaumeGomez** [comment]
If the doc comments come from somewhere under LGPL, then LGPL, otherwise it's MIT and you can leave them where they are.

File: src/image_surface.rs [hunk 1]
```diff
@@ -48,23 +49,26 @@ impl ImageSurface {
...
-            }
+        let result = unsafe {
+            ImageSurface::from_raw_full(
+                ffi::cairo_image_surface_create_for_data(ptr, format.into(), width, height, stride)
+            )
+        };
+        if let Ok(surface) = &result {
+            static IMAGE_SURFACE_DATA: crate::UserDataKey<Box<dyn AsMut<[u8]>>> =
+                crate::UserDataKey::new();
+            surface.set_user_data(&IMAGE_SURFACE_DATA, Rc::new(data))
```
Comment (sdroege) [inline src/image_surface.rs]: Why does this have to be in an `Rc`?

Comment (sdroege) [reply src/image_surface.rs]: Oh I see, because of the getter

**Comment by sdroege** [comment]
:+1:

**Comment by GuillaumeGomez** [comment]
Thanks @SimonSapin (especially for the documentation!)!

**Pull request merged**